### ISSUE TITLE
no-extra-parens have false positive about for-of statement

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#9755]
**Tell us about your environment**

* **ESLint Version:** v4.13.1
* **Node Version:** -
* **npm Version:** -

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**
**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

> ```js
> /*eslint no-extra-parens: [error, all] */
>
> for (a in (b, c));
> for (a of (b, c));
> ```
> [Online demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZXh0cmEtcGFyZW5zOiBbZXJyb3IsIGFsbF0gKi9cclxuXHJcbmZvciAoYSBpbiAoYiwgYykpO1xyXG5mb3IgKGEgb2YgKGIsIGMpKTtcclxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**What did you expect to happen?**

ESLint reports an error:

- `3:11 - Gratuitous parentheses around expression. (no-extra-parens)`

Because `for (a of b, c);` is syntax error. (related to https://github.com/ternjs/acorn/issues/642)

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reports 2 errors:

- `3:11 - Gratuitous parentheses around expression. (no-extra-parens)`
- `4:11 - Gratuitous parentheses around expression. (no-extra-parens)`

**Comment by erindepew** [comment]
@mysticatea if this issue hasn't already been claimed I'd be interested in fixing this!

**Comment by mysticatea** [comment]
@erindepew Please! I have not started yet.
